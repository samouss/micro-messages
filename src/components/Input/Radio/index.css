.Radio {
  display: flex;
}

.Radio:first-child {
  margin-right: 10px;
}

.Radio:hover {
  cursor: pointer;
}

.Radio__Input {
  margin-right: 5px;
}
