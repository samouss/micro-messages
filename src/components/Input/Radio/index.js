// @flow

import React from 'react';
import './index.css';

type Props = {
  id: string,
  children: React.Element<*>,
};

const Radio = ({ id, children, ...props }: Props) => (
  <label
    htmlFor={id}
    styleName="Radio"
  >
    <input
      id={id}
      styleName="Radio__Input"
      type="radio"
      {...props}
    />

    {children}
  </label>
);

export default Radio;
