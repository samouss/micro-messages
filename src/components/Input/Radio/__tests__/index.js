// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Radio from '../index';

describe('<Radio />', () => {
  const defaultProps = {
    id: 'input',
    value: 'visible',
    onChange: () => {},
  };

  it('expect to render the component correctly', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Radio
        {...props}
      >
        <span>some content</span>
      </Radio>,
    );

    expect(component).toMatchSnapshot();
  });
});
