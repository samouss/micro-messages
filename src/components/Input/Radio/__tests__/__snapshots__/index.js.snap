// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> expect to render the component correctly 1`] = `
<label
  styleName="Radio"
>
  <input
    onChange={[Function]}
    styleName="Radio__Input"
    type="radio"
    value="visible"
  />
  <span>
    some content
  </span>
</label>
`;
